### ISSUE TITLE
how can i make my custom model get the high map as same as pretrained model yolov8n released?

### ISSUE THREAD

**Issue by cheng-lao** [ultralytics/ultralytics#17150]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

hi, thanks for your excellent work! 
 
I would like to ask a question about a modification in the yolov8 model. When I was training, I found that the accuracy of the network I trained using the [official yolov8 yaml file configuration file](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/models/v8/yolov8.yaml) was always nearly 10 maps lower than the actual accuracy. And the same situation happened in my modified yolo yaml file ,that when i trained my slightly modified model on the coco datasets.  i got the map was nearly 10 maps lower than the actual accuracy. 

 However, the effect of using [yolov8 the pre-trained weights](https://github.com/ultralytics/assets/releases/download/v8.2.0/yolov8n.pt) for training is similar to the official test accuracy.**I would like to ask if this is because the official has made fine adjustments to the pre-trained weights?**

That is, because the official parameters are adjusted well during training, the effect of the pre-trained model is better than the effect obtained from training from scratch. **If this is the case, how can I make the modified model trained from scratch achieve the same good results as the official pre-trained model? Or even better?**

Here is  my modified model 👇：
```
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLOv8 object detection model with P3-P5 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect

# Parameters
nc: 80 # number of classes
scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024] # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
  s: [0.33, 0.50, 1024] # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
  m: [0.67, 0.75, 768] # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
  l: [1.00, 1.00, 512] # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
  x: [1.00, 1.25, 512] # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2        # 0
  - [-1, 1, Conv, [128, 3, 2]] # 1-P2/4       # 1  
  - [-1, 3, C2f, [128, True]]                 # 2
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8       # 3
  - [-1, 6, C2f, [256, True]]                 # 4
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16      # 5
  - [-1, 6, C2f, [512, True]]                 # 6
  - [-1, 1, Conv, [1024, 3, 2]] # 7-P5/32     # 7
  - [-1, 3, C2f, [1024, True]]                # 8
  - [-1, 1, SPPF, [1024, 5]] # 9

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, C3k2, [512, False]] # 12                      modify here 👈

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, C3k2, [256, False]] # 15 (P3/8-small)                      modify here 👈

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]] # cat head P4
  - [-1, 3, C3k2, [512, False]] # 18 (P4/16-medium)                      modify here 👈

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]] # cat head P5
  - [-1, 3, C3k2, [1024, True]] # 21 (P5/32-large)                      modify here 👈

  - [[15, 18, 21], 1, Detect, [nc]] # Detect(P3, P4, P5)

```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cheng-lao, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response, but rest assured, an Ultralytics engineer will be along shortly to assist you further.

We recommend checking out our [Documentation](https://docs.ultralytics.com) where you can find extensive guides and examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. This is a great resource to address some common questions new users might have and can provide some immediate insights into your query.

If your issue is related to training discrepancies with custom models, we suggest providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This will help us understand the problem better and assist you in debugging the issue more effectively.

To enhance your custom training performance and potentially match or exceed pretrained model metrics, please ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Including dataset examples and training logs can also be very helpful.

For real-time support, join our vibrant [Discord community](https://ultralytics.com/discord) 🎧. In-depth discussions can be had on [Discourse](https://community.ultralytics.com), or if you prefer, engage with like-minded peers on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please make sure you’ve upgraded to the latest `ultralytics` package. This ensures you have the latest features and fixes:

```bash
pip install -U ultralytics
```

YOLO can be run in the following environments with all dependencies preinstalled. Choose one that suits you:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Check out our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Refer to our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Instructions are available in our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. This ensures correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different operating systems.

**Comment by Skillnoob** [comment]
I don't think the pretrained models use the same hyperparameters as the defaults.
You can check that with the following code:
```python
from ultralytics import YOLO

model = YOLO('yolov8n.pt')

print(model.ckpt["train_args"])
```

**Comment by cheng-lao** [comment]
> I don't think the pretrained models use the same hyperparameters as the defaults. You can check that with the following code:
> 
> ```python
> from ultralytics import YOLO
> 
> model = YOLO('yolov8n.pt')
> 
> print(model.ckpt["train_args"])
> ```

ok , i try the code and get the follow output! And i bolded the difference between the output and [the official default.yaml](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml)

{'task': 'detect', 'mode': 'train', 'model': 'yolov8n.yaml', 'data': 'coco.yaml', **'epochs': 500**, **'patience': 50**, 'batch': 16, 'imgsz': 640, 'save': True, 'cache': 'disk', 'device': 0, 'workers': 8, 'project': 'YOLOv8', 'name': 'yolov8n', 'exist_ok': False, **'pretrained': False**, **'optimizer': 'SGD'**, 'verbose': False, 'seed': 0, 'deterministic': True, 'single_cls': False, **'image_weights': False**, 'rect': False, 'cos_lr': False, 'close_mosaic': 10, 'resume': False, 'overlap_mask': True, 'mask_ratio': 4, 'dropout': False, 'val': True, 'save_json': False, 'save_hybrid': False, **'conf': 0.001**, 'iou': 0.7, 'max_det': 300, **'half': True**, 'dnn': False, 'plots': True, 'source': 'ultralytics/assets/', 'show': False, 'save_txt': False, 'save_conf': False, 'save_crop': False, 'hide_labels': False, 'hide_conf': False, 'vid_stride': 1,**'line_thickness': 3**, 'visualize': False, 'augment': False, 'agnostic_nms': False, 'retina_masks': False, 'format': 'torchscript', 'keras': False, 'optimize': False, 'int8': False, 'dynamic': False, 'simplify': False, 'opset': 17, 'workspace': 4, 'nms': False, 'lr0': 0.01, 'lrf': 0.01, 'momentum': 0.937, **'weight_decay': 0.001**, 'warmup_epochs': 3.0, 'warmup_momentum': 0.8, 'warmup_bias_lr': 0.1, 'box': 7.5, 'cls': 0.5, 'dfl': 1.5, **'fl_gamma': 0.0**, 'label_smoothing': 0.0, 'nbs': 64, 'hsv_h': 0.015, 'hsv_s': 0.7, 'hsv_v': 0.4, 'degrees': 0.0, 'translate': 0.1, 'scale': 0.5, 'shear': 0.0, 'perspective': 0.0, 'flipud': 0.0, 'fliplr': 0.5, 'mosaic': 1.0, 'mixup': 0.0, 'copy_paste': 0.0, 'cfg': None, 'v5loader': True}

i bolded the difference between the output and the official yolov8 yaml file configuration file. And i think the main difference is the `epochs==500` !!?? Previously , i trained my model with the `epochs==300`, so does the low epochs make my result not ideal ?

(`weight_decay` and `optimizer` are `0.0005` and `auto` respectively in the `default.yaml` . So i think they both has a slightly influence in training model.)


So the output from the code is the training hyperparameters?  And the main cause to my unexpected training output  is the my epochs is so low?  

(sorry , i just want to confirm , i am a beginner)

**Comment by Skillnoob** [comment]
The v8 pretrained models are trained with 500 epochs, yes. Some of the parameters you highlighted are determined automatically based on epochs and dataset size, as well as class count.

**Comment by cheng-lao** [comment]
> The v8 pretrained models are trained with 500 epochs, yes. Some of the parameters you highlighted are determined automatically based on epochs and dataset size, as well as class count.

So if I train the yolov8 model with epochs==500 using the yaml file, I can achieve the same results as the pre-trained model?

the train code as below:
```python
from ultralytics import YOLO


model = YOLO(r'ultralytics\cfg\models\v8\yolov8.yaml')

train_result = model.train(
    data = "coco.yaml",
    epochs = 500,
    optimizer = 'SGD',
    weight_decay = 0.001,
)

model.val()

```

**Comment by Skillnoob** [comment]
When using val you should also supply coco.yaml to it, since by default val uses coco128. You should hopefully get similar results, but there may be other factors that are different and could influence the results.

**Comment by cheng-lao** [comment]
> When using val you should also supply coco.yaml to it, since by default val uses coco128. You should hopefully get similar results, but there may be other factors that are different and could influence the results.

ok! ok!. i will add the `data='coco.yaml'` , i have got the enough information to answer my question . i will try trainging the model with `epochs==500`  again , and if i encounter other difficulties , i will ask again .

thanks ! 🥰

**Comment by cheng-lao** [comment]
> When using val you should also supply coco.yaml to it, since by default val uses coco128. You should hopefully get similar results, but there may be other factors that are different and could influence the results.

wait, i want to ask what factors will affect the results?

**Comment by Skillnoob** [comment]
For example the current yolov8 head is now the yolo11 head, since `Detect` got changed in the process of yolo11 being developed. The "new" yolov8 should probably perform slightly better than the original v8.

**Comment by cheng-lao** [comment]
oh, i take it.
I'd better try it first. thanks again !💖

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
